### PR TITLE
Update permission command to use args, not options

### LINKED ISSUE #675: Redesign `permissions` command
Permissions needs to be reworked away from long flags similar to `groups` in favor of subcommands and positional args.

```
USAGE

permissions <subcommand>

permissions list
permissions create <site:name>
permissions delete <site:name>
permissions grant <namespace:name> <role-names ...>
permissions revoke <namespace:name> <role-name ...>

EXAMPLES

permissions list
> NAME                  ROLES
  site:deploy-frontend  ops ux
  site:deploy-backend   ops

permissions create site:deploy
> Created permission site:deploy

permissions delete site:deploy
> Deleted permission site:deploy

permissions grant site:deploy ops
> Granted permission site:deploy to role ops

permissions grant site:deploy ops ux
> Granted permission site:deploy to roles ops ux

permissions revoke site:deploy ops
> Revoked permission site:deploy from role ops

permissions revoke site:deploy ops ux
> Revoked permission site:deploy from roles ops ux
```

### PR DESCRIPTION
**Pull request by christophermaier**
Boldly, into the future!

Fixes #675

### CODE REVIEW

File: lib/cog/commands/helpers.ex [hunk 1]
```diff
@@ -150,4 +152,20 @@ defmodule Cog.Commands.Helpers do
...
+                                                   String.replace(acc, "%{#{key}}", to_string(value))
+                                                 end)
+                                               msg ->
+                                                 msg
+                                             end)
+
+    msg_map
+    |> Enum.map(fn({field, msg}) -> "#{field} #{msg}" end)
+    |> Enum.join("\n")
+  end
```
Comment (vanstee) [inline lib/cog/commands/helpers.ex]: Really wish Phoenix came with a function that does this. Had to write something similar for Flywheel.

File: test/support/adapter_assertions.ex [hunk 2]
```diff
@@ -23,7 +23,14 @@ defmodule Cog.AdapterAssertions do
...
     |> String.replace(~r/^}/m, "},")
     |> String.rstrip(?,)
-    Poison.decode!("[#{response}]", keys: :atoms)
+
+    case Poison.decode("[#{response}]", keys: :atoms) do
+      {:ok, decoded} ->
+        decoded
+      {:error, _} ->
+        # This makes for easier test debugging
+        raise "Expected JSON, got: #{inspect response}"
```
Comment (vanstee) [inline test/support/adapter_assertions.ex]: :+1:

**Comment by vanstee** [comment]
LGTM. Do we have an issue to double check that all `permission` command references are updated in the docs?

**Comment by christophermaier** [comment]
@vanstee That's a good call... I'll make one.

**Pull request merged**